hedgehog/src/Hedgehog/Internal/Gen.hs

```diff
@@ -522,6 +524,16 @@ instance (MonadGen m, Monoid w) => MonadGen (Strict.RWST r w s m) where
 ------------------------------------------------------------------------
 -- GenT instances
 
+-- technically, the Monoid constraint on @a@ could be weakened
+-- to Semigroup, but this would be annoying for older GHCs
+-- where Semigroup is not a superclass of Monoid.
+instance (Monad m, Monoid a) => Semigroup (GenT m a) where
```

Comment:
I would prefer that you weaken the constraint to `Semigroup a` and delete the comment.

>this would be annoying for older GHCs

Not if the user depends on the `semigroups` package, as [I've written about](https://qfpl.io/posts/semigroups/). It provides `Semigroup` compatibility at least as far back as GHC 7.2.

On the flip-side, the current approach would be unhelpful to any user whose type is a `Semigroup` but not a `Monoid`.

I have made this change on my fork [here](https://github.com/gwils/haskell-hedgehog/commit/4e134f7f42365195568376eda7288fc5b318c738) if you'd like to use that patch.

Reply:
Yeah, George's version looks right to me

Reply:
Good call. I had originally thought this wouldn't work with pre-SMP `base`, but it totally does. This is certainly the best option. I merged it into the PR.